test/src/tests/acceptancetests/paginggrouptest/paging_test.go

```diff
@@ -29,209 +29,175 @@ func getPagingClient() PagingClient {
+	count := 0
+	for page, err := pagingClient.GetMultiplePagesRetrySecond(context.Background()); page.NotDone(); err = page.Next() {
 		c.Assert(err, chk.IsNil)
-		res = resNext
+		c.Assert(page.Values(), chk.NotNil)
+		count++
 	}
 	c.Assert(count, chk.Equals, 10)
 
-	resChan, errChan := pagingClient.GetMultiplePagesRetrySecondComplete(context.Background())
 	count = 0
-	for item := range resChan {
+	for iter, err := pagingClient.GetMultiplePagesRetrySecondAll(context.Background()); iter.NotDone(); err = iter.Next() {
+		c.Assert(err, chk.IsNil)
+		c.Assert(iter.Value().Properties, chk.NotNil)
 		count++
-		c.Assert(item, chk.NotNil)
 	}
 	c.Assert(count, chk.Equals, 10)
-	c.Assert(<-errChan, chk.IsNil)
 }
 
 func (s *PagingGroupSuite) TestGetSinglePagesFailure(c *chk.C) {
-	res, err := pagingClient.GetSinglePagesFailure(context.Background())
+	page, err := pagingClient.GetSinglePagesFailure(context.Background())
 	c.Assert(err, chk.NotNil)
-	c.Assert(res.StatusCode, chk.Equals, http.StatusBadRequest)
+	c.Assert(page.Response().StatusCode, chk.Equals, http.StatusBadRequest)
 
-	resChan, errChan := pagingClient.GetSinglePagesFailureComplete(context.Background())
 	count := 0
-	for item := range resChan {
+	for iter, err := pagingClient.GetSinglePagesFailureAll(context.Background()); iter.NotDone(); err = iter.Next() {
+		c.Assert(err, chk.IsNil)
+		c.Assert(iter.Value().Properties, chk.NotNil)
 		count++
-		c.Assert(item, chk.NotNil)
 	}
 	c.Assert(count, chk.Equals, 0)
-	c.Assert(<-errChan, chk.NotNil)
 }
 
 func (s *PagingGroupSuite) TestGetMultiplePagesFailure(c *chk.C) {
-	res, err := pagingClient.GetMultiplePagesFailure(context.Background())
+	page, err := pagingClient.GetMultiplePagesFailure(context.Background())
 	c.Assert(err, chk.IsNil)
-	c.Assert(res.NextLink, chk.NotNil)
-	res, err = pagingClient.GetMultiplePagesFailureNextResults(res)
+	c.Assert(page.NotDone(), chk.Equals, true)
+	err = page.Next()
 	c.Assert(err, chk.NotNil)
-	c.Assert(res.StatusCode, chk.Equals, http.StatusBadRequest)
+	//c.Assert(page.Response().StatusCode, chk.Equals, http.StatusBadRequest)
 
-	resChan, errChan := pagingClient.GetMultiplePagesFailureComplete(context.Background())
 	count := 0
-	for item := range resChan {
+	for iter, err := pagingClient.GetMultiplePagesFailureAll(context.Background()); err == nil; err = iter.Next() {
+		c.Assert(err, chk.IsNil)
+		c.Assert(iter.Value().Properties, chk.NotNil)
 		count++
-		c.Assert(item, chk.NotNil)
 	}
 	c.Assert(count, chk.Equals, 1)
-	c.Assert(<-errChan, chk.NotNil)
 }
 
 func (s *PagingGroupSuite) TestGetMultiplePagesFailureURI(c *chk.C) {
-	res, err := pagingClient.GetMultiplePagesFailureURI(context.Background())
+	page, err := pagingClient.GetMultiplePagesFailureURI(context.Background())
 	c.Assert(err, chk.IsNil)
-	c.Assert(*res.NextLink, chk.Equals, "*&*#&$")
-	_, err = pagingClient.GetMultiplePagesFailureURINextResults(res)
+	//c.Assert(*res.NextLink, chk.Equals, "*&*#&$")
+	err = page.Next()
```

Comment:
Is there a way to easily get the next link?

Reply:
Not at present.  I can add a NextLink() method to the generated code if you think it adds value.

Reply:
Thinking about this more, I want to change the generated Response() method to return the response object, not just the autorest.Response.

Reply:
Fixed.